ch_trees/gui.py

```diff
@@ -57,77 +63,313 @@ class TreeGen(bpy.types.Operator):
     render_output_path = os.path.sep.join((os.path.expanduser('~'), 'treegen_render.png'))
-    bpy.types.Scene.render_output_path_input = bpy.props.StringProperty(name="", default=render_output_path)
+    _scene.render_output_path_input = _props.StringProperty(name="", default=render_output_path)
+
+    # ======================
+    # Tree customizer inputs
+
+    # Tree Shape
+    tree_shape_options = (
+        ('0', 'Conical', 'Conical'), ('1', 'Spherical', 'Spherical'), ('2', 'Hemispherical', 'Hemispherical'),
+        ('3', 'Cylindrical', 'Cylindrical'), ('4', 'Tapered Cylindrical', 'Tapered Cylindrical'),
+        ('5', 'Flame', 'Flame'), ('6', 'Inverse Conical', 'Inverse Conical'), ('7', 'Tend Flame', 'Tend Flame')
+    )
+    _scene.tree_shape_input = _props.EnumProperty(name="", items=tree_shape_options, default='0')
+
+    _scene.tree_prune_ratio_input = _props.FloatProperty(name="", default=0, min=0, max=1)
+    _scene.tree_prune_width_input = _props.FloatProperty(name="", default=.5, min=.000001, max=200)
+    _scene.tree_prune_width_peak_input = _props.FloatProperty(name="", default=.5, min=0, max=200)
+    _scene.tree_prune_power_low_input = _props.FloatProperty(name="", default=.5, min=-200,
+                                                                         max=200)  # <1 convex, >1 concave
+    _scene.tree_prune_power_high_input = _props.FloatProperty(name="", default=.5, min=-200,
+                                                                          max=200)  # <1 convex, >1 concave
+
+    # Size of base
+    _scene.tree_base_size_input = _props.FloatVectorProperty(name="", default=(0.3, 0.02, 0.02, 0.02), size=4, min=.001)
+
+    # Floor split count
+    _scene.tree_floor_splits_input = _props.IntProperty(name="", default=0, min=0, max=500)
+
+    # Base split count
+    _scene.tree_base_splits_randomize_input = _props.BoolProperty(name="Randomize base split count", default=False)
+    _scene.tree_base_splits_input = _props.IntProperty(name="", default=0, min=-5, max=5)
+    _scene.tree_base_splits_limit_input = _props.IntProperty(name="", default=0, min=0, max=10)
+
+    # Overall tree scale and scale variation
+    _scene.tree_g_scale_input = _props.FloatProperty(name="", default=13, min=.000001, max=150)
+    _scene.tree_g_scale_v_input = _props.FloatProperty(name="", default=3, min=0, max=149.99)
+
+    # Level count
+    _scene.tree_levels_input = _props.IntProperty(name="", default=3, min=1, max=6)
+
+    # Ratio and ratio power
+    _scene.tree_ratio_input = _props.FloatProperty(name="", default=.015, min=.000001, max=1)
+    _scene.tree_ratio_power_input = _props.FloatProperty(name="", default=1.2, min=0, max=5)
+
+    # Flare
+    _scene.tree_flare_input = _props.FloatProperty(name="", default=.6, min=0, max=10)
+
+    # Branch appearance
+    _scene.tree_branches_input = _props.FloatVectorProperty(name="", default=(-.01, 50, 30, 10), size=4, min=-.01, max=360)
```

Comment:
Any reason -.01 is used a default/minimum for many parameters - I thought I used 0 as a default in most cases?

Reply:
It used -0 as default, which Python can't represent. I used -.01 as a workaround for "negative close-ish to 0", but if +0 will work that can be changed